### PR TITLE
Add BAMOE staging job and fix potential bug in current rhba staging job

### PR DESCRIPTION
**Pull request by AntStephenson**
I've created a temp job here to test it https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/PROD/job/rhba-staging-bamoe

### CODE REVIEW

File: job-dsls/src/main/resources/job-scripts/prod_rhba_staging.jenkinsfile [hunk 1]
```diff
@@ -136,8 +136,10 @@ def createStagingPath(path) {
 }
 
 def stageArtifact(url, path) {
-    println "[INFO] staging ${url} to ${path}"
-    remoteExec("curl -s ${url} -o ${path}")
+    // need to use indy-admin here as there's a caching bug which can make some downloads fail
+    def indyAdmin = url.replace("indy", "indy-admin")
```
Comment (rgdoliveira) [inline job-dsls/src/main/resources/job-scripts/prod_rhba_staging.jenkinsfile:140]: Just out of curiosity, is "indy" always expected as part of the url? if it is not in the url is the replace going to fail and stop the execution?

Comment (AntStephenson) [reply job-dsls/src/main/resources/job-scripts/prod_rhba_staging.jenkinsfile:140]: yes, indy is always expected here in the download url

**Review by rgdoliveira — APPROVED** [review]
LGTM

**Review by Ginxo — APPROVED** [review]
LGTM

**Pull request merged**